BWAPI4J/src/main/java/org/openbw/bwapi4j/BW.java

```diff
@@ -240,14 +237,16 @@ private void extractBridgeDependencies(final BridgeType bridgeType) {
     }
   }
 
-  private static void extractFileIfNotExists(
-      final ZipFile zipFile, final String sourceFilename, final String targetDirectory)
-      throws ZipException {
-    final Path targetFile = Paths.get(targetDirectory, sourceFilename);
-    if (!Files.isRegularFile(targetFile)
-        && !Files.isDirectory(targetFile)
-        && !Files.exists(targetFile)) {
-      zipFile.extractFile(sourceFilename, targetDirectory);
+  private static void extractFileIfNotExists(final String sourceFilename, final String targetDirectory) {
+    final File library = new File(targetDirectory, sourceFilename);
+    if (!library.exists()) {
+      InputStream is = BW.class.getResourceAsStream(File.separator + sourceFilename);
+      try {
+        Files.copy(is, library.toPath(), StandardCopyOption.REPLACE_EXISTING);
+      } catch (IOException e) {
```

Comment:
What is the justification for catching an `IOException` and not rethrowing it? This makes debugging more difficult as the library will fail to load and we won't know exactly why. Here, we would be able to see it's not extracting properly. It should be thrown in some way. It could be wrapped in an runtime exception. E.g. `throw new RuntimeException(e);`

Reply:
The error can be re-thrown on cleanup, but I think I'll remove this since I'd have to test external jar first.